### PR TITLE
Added an example of failure with an entity containing a collection of discriminated entities

### PR DESCRIPTION
**Pull request by ymarillet**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | no
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

Hi there,

I've got 2 problems here, both of them regarding a collection of discriminated entities. I've tried to keep the example as simple as possible

* first, running the new behat test provided (behat features/main/discr.feature -v), we can see that api platform assumes that a collection contains only a certain type of elements, but that's not 100% correct in the case of a collection of discriminated entities. It results that the entire collection's elements inherit a single PropertyMetadata in the AbstractItemNormalizer

* second, if we remove the serializer group "discr_container_dummy" from the DiscrAbstractDummy entity, we can see that the test throws an exception because the parent denormalization group could not be found (again AbstractItemNormalizer)

Any idea ?

Regards,
Yohann

### CODE REVIEW

File: tests/Fixtures/TestBundle/Serializer/Denormalizer/DiscrAbstractDummyDenormalizer.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
+<?php
```
Comment (dunglas) [inline tests/Fixtures/TestBundle/Serializer/Denormalizer/DiscrAbstractDummyDenormalizer.php]: This class should not be necessary. IIRC, API Platform is (or should be) able to automatically detect the type to use for normalization.

Comment (ymarillet) [inline tests/Fixtures/TestBundle/Serializer/Denormalizer/DiscrAbstractDummyDenormalizer.php]: The problem resides in the fact that the collection is a collection of DiscrAbstractDummy (which is, abstract) and thus cannot be instancied directly. This require a similar functionality as in the JMS serializer (or doctrine) to be implemented in the SF serializer to be able to denormalize correctly out of the box

Comment (dunglas) [inline tests/Fixtures/TestBundle/Serializer/Denormalizer/DiscrAbstractDummyDenormalizer.php]: IMO `abstract` classes are a bad idea when dealing with API Platform, and with Doctrine generally speaking.

However this specific case should be handle automatically, we even have a test for that: https://github.com/api-platform/core/blob/master/features/main/crud_abstract.feature

Comment (ymarillet) [inline tests/Fixtures/TestBundle/Serializer/Denormalizer/DiscrAbstractDummyDenormalizer.php]: Again, the problem resides in the fact that we cannot identify the type of an element within the collection without providing that information, that's why the supportsDenormalization method checks for a "@type" property in the provided data ...
Is it a "won't fix" issue ?

**Comment by dunglas** [comment]
I need to dive into this problem, but if we find an appropriate fix, we need to merge it!

**Comment by soyuka** [comment]
Is this still an issue? Closing because of old lmk if I should reopen.

**Pull request closed without merging**